dubbo-common/src/main/java/org/apache/dubbo/common/logger/log4j/Log4jLoggerAdapter.java

```diff
@@ -26,11 +26,32 @@
 
 import java.io.File;
 import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
 
 public class Log4jLoggerAdapter implements LoggerAdapter {
 
     private File file;
 
+    private static final Map<Level, org.apache.log4j.Level> toLevel = new HashMap<>();
+    private static final Map<org.apache.log4j.Level, Level> toApacheLevel = new HashMap<>();
+
+    static {
+        toLevel.put(Level.ALL, org.apache.log4j.Level.ALL);
+        toLevel.put(Level.TRACE, org.apache.log4j.Level.TRACE);
+        toLevel.put(Level.DEBUG, org.apache.log4j.Level.DEBUG);
+        toLevel.put(Level.INFO, org.apache.log4j.Level.INFO);
+        toLevel.put(Level.WARN, org.apache.log4j.Level.WARN);
+        toLevel.put(Level.ERROR, org.apache.log4j.Level.ERROR);
+
+        toApacheLevel.put(org.apache.log4j.Level.ALL, Level.ALL);
+        toApacheLevel.put(org.apache.log4j.Level.TRACE, Level.TRACE);
+        toApacheLevel.put(org.apache.log4j.Level.DEBUG, Level.DEBUG);
+        toApacheLevel.put(org.apache.log4j.Level.INFO, Level.INFO);
+        toApacheLevel.put(org.apache.log4j.Level.WARN, Level.WARN);
+        toApacheLevel.put(org.apache.log4j.Level.ERROR, Level.ERROR);
```

Comment:
This may cause `NoClassDefFoundError` if log4j's dependency is absent.

Reply:
you re right